plugin-base/src/main/kotlin/io/spine/tools/gradle/protobuf/ProtobufGradlePluginAdapter.kt

```diff
@@ -0,0 +1,214 @@
+ */
+private class AdapterImpl(
+    project: Project,
+    private val delegate: ProtobufGradlePluginAdapter = createDelegate(project)
+) : ProtobufGradlePluginAdapter by delegate {
+
+    companion object {
+        fun createDelegate(project: Project): ProtobufGradlePluginAdapter {
+            val newApi = NewApi.findExtension(project) != null
+            return if (newApi) NewApi(project) else ConvApi(project)
+        }
+    }
+}
+
+/**
+ * Adapter for the API of Protobuf Gradle Plugin after v0.9.0.
+ */
+private class NewApi(override val project: Project): ProtobufGradlePluginAdapter {
+
+    private val extension: Any = findExtension(project)!!
+    private val extensionClass: Class<*> = extension.javaClass
+
+    override var generatedFilesBaseDir: String
+        get() {
+            val getter = extensionClass.getMethod("getGeneratedFilesBaseDir")
+            return getter.invoke(extension) as String
+        }
+
+        set(value) {
+            val setter = extensionClass.getMethod("setGeneratedFilesBaseDir", String::class.java)
+            setter.invoke(extension, value)
+        }
+
+    override fun protoc(action: Action<ExecutableLocator>) {
+        val setter = extensionClass.getMethod("protoc")
+        setter.invoke(extension, action)
+    }
+
+    override fun plugins(action: Action<NamedDomainObjectContainer<ExecutableLocator>>) {
+        val pluginsMethod = extensionClass.getMethod("plugins", Action::class.java)
+        pluginsMethod.invoke(extension, action)
+    }
+
+    override fun configureProtoTasks(action: Action<GenerateProtoTask>) {
+        val generateProtoTasks = extensionClass.getMethod("generateProtoTasks", Action::class.java)
+
+        // The actual type on which the closure operates (in newer API) is
+        // `ProtobufExtension.GenerateProtoTaskCollection`. We cannot use this type because
+        // it may not be available if older Gradle plugin is applied. We'd have
+        // `ProtobufConfigurator.GenerateProtoTaskCollection` instead. Therefore, we obtain
+        // an instance of `all()` tasks via a reflection call on the closure argument.
+        val closure = closure<Any> {
+            val all = it.javaClass.getMethod("all")
+            val allTasks = all.invoke(it)
+            @Suppress("UNCHECKED_CAST")
+            (allTasks as TaskCollection<GenerateProtoTask>).forEach { task ->
+                action.execute(task)
+            }
+        }
+        // Now pass the closure for the Protobuf Gradle plugin for being applied later.
+        generateProtoTasks.invoke(extension, closure)
+    }
+
+    companion object {
+        const val extensionName = "protobuf"
+
+        fun findExtension(project: Project): Any? =
+            project.extensions.findByName(extensionName)
+    }
+}
+
+/**
+ * Adapter for the API of Protobuf Gradle Plugin pre v0.9.0.
+ */
+private class ConvApi(override val project: Project): ProtobufGradlePluginAdapter {
```

Comment:
Does "Conv" stand for `Conventional`?

I would expect something like `LegacyApi`.